### ISSUE TITLE
 [Fast Furnaces] Failed to load mod due to productivitymodulelimitation

### ISSUE THREAD

**Issue by Funketime** [ReikaKalseki/Reika_FactorioMods_Issues#166]
I received this error message after upgrading Factorio and first loading of Fast Furnaces:

Error loading mods 
Failed to load mods: _ _FastFurnaces_ _/data-updates.lua:31: _ _FastFurnaces_ _ /prototypes/general-updates.lua:9: attempt to call global 'productivitymodulelimitation' (a nil value)
stack traceback: 
_ _ FastFurnaces_ _ /prototypes/general-updates.lua:9: in main chunk
[C]: in function 'require' 
_ _ FastFurnaces_ _ /data-updates.lua:31: in main chunk 
stack traceback: 
[C]: in function 'require' 
_ _ FastFurnaces_ _ _ /data-updates.lua:31: in main chunk 
Mods to be disabled: 
• FastFurnaces

I am using Fast Furnaces 1.17.6 and Factorio 0.17.48.  I have Dragon Industries 1.17.6 loaded.

**Comment by Funketime** [comment]
I duplicated the error after I unloaded every other mod.  However, I was able to get it to load without error by changing editing setting.lua by changing default_value  in free-use-modules from false to true.

To test this fix I deleted the mod and reinstalled without my change it and it ran without error.  I cannot duplicate the error anymore.

There is no error anymore and nothing I can do seems to be able to initiate it so consider it solved.

**Comment by ReikaKalseki** [comment]
I already fixed this in code.

**Comment by Funketime** [comment]
Nice work. Thank you for the great mods and timely fixes. 

I found out why I could not duplicate the error,  There is a factorio file called mod-settings.dat that maintains the setting even of mods that are deleted.  So when I made changes to the default settings then tried to repeat the error, this file would retain the settings.  When I deleted this file, I was able to duplicate the error.  I then got a new error if I enabled the fast refinery and if the free-use-modules was set to true.  I have assumed that it is related to this error as it is another nil error so will not post a new bug until I see the update.
![refinery](https://user-images.githubusercontent.com/51725096/59469862-3bb81580-8df3-11e9-9a12-60e9132ab7ab.png)
